[Docs] task distribution (#5246)
recognai/rubrix: argilla/docs/how_to_guides/distribution.md

---
description: In this section, we will provide a step-by-step guide to show how to distribute the annotation task among team members.
---

# Distribute the task among the team

This guide explains how you can use Argilla’s **automatic task distribution** to divide the task of annotating a dataset among multiple team members efficiently.

> Users can define the minimum number of submitted responses expected for each record depending on whether the dataset should have annotation overlap and how much. Argilla will use this setting to handle automatically the records that will be shown in the pending queues of the annotation team. When a record has met the minimum number of submissions, the status of the record will change to `completed` and the record will be removed from the `Pending` queue of all team members, so they can focus on providing responses where they are most needed. The dataset’s annotation task will be fully completed once all records have the `completed` status.

Reviewer: ```suggestion
Users can define the minimum number of submitted responses expected for each record depending on whether the dataset should have annotation overlap and how much. Argilla will use this setting to handle the records automatically that will be shown in the pending queues of the annotation team. When a record has met the minimum number of submissions, the status of the record will change to `completed` and the record will be removed from the `pending` queue of all team members, so they can focus on providing responses where they are most needed. The dataset’s annotation task will be fully completed once all records have the `completed` status.
```

!!! note

>     The status of a record can be either `completed`, when it has the required number of responses with `submitted` status, or `left`, when it doesn’t meet this requirement.

Reviewer: this is a bit vague perhaps we can redirect them to the introduction of the different statuses or maybe a table would be clearer?

Reviewer: because we've got records status, responses status etc, so I think it would be good to have an overview or mention both of them. I would say, having a tables probably covers his best.

Reviewer 2: I agree, for me, the differences are a bit unclear

    Each record can have multiple responses and each of those can have the status `submitted`, `discarded` or `draft`.

!!! info "Main Class"

    ```python
    rg.Dataset(
        name="name",
        workspace="workspace",
        settings=settings,
        client=client
    )

>     ```

Reviewer: this code does not show how to use the task distribution class but rather the datasets class

---
description: In this section, we will provide a step-by-step guide to show how to distribute the annotation task among team members.
---

> # Distribute the annotation among the team

Reviewer 2: Unsure about the title here too

Author: Do you think "Distribute the annotation task among the team" makes more sense to you?

> This guide explains how you can use Argilla’s **automatic task distribution** to divide the task of annotating a dataset among multiple team members efficiently.

Reviewer 2: ```suggestion
This guide explains how you can use Argilla’s **automatic task distribution** to efficiently divide the task of annotating a dataset among multiple team members.
```

> Users can define the minimum number of submitted responses expected for each record depending on whether the dataset should have annotation overlap and how much. Argilla will use this setting to handle automatically the records that will be shown in the pending queues of all users with access to the dataset.

Reviewer 2: ```suggestion
Users can set the minimum number of responses expected for each record based on the desired level of annotation overlap for the dataset. Argilla will use this setting to automatically handle the records that will be shown in the pending queues of all users with access to the dataset.
```

!!! tip
    Leave the default value of minimum submissions (1) if you are working on your own or when you don't require more than one submitted response per record.

If you wish to set a different number, you can do so through the `distribution` setting in your dataset settings:

```python
settings = rg.Settings(
    guidelines="These are some guidelines.",
    fields=[
        rg.TextField(
            name="text",
        ),
    ],
    questions=[
        rg.LabelQuestion(
            name="label",
            labels=["label_1", "label_2", "label_3"]
        ),
    ],
    distribution=rg.TaskDistribution(min_submitted=3)
)
```

> Learn more about configuring dataset settings in the [Dataset management guide](../how_to_guides/dataset.md).

!!! tip
    Increase the number of minimum subsmissions if you’d like to ensure you get more than one submitted response per record. Make sure that this number is never higher than the number of members in your team. Note that the lower this number is, the faster the task will be completed.

!!! note

>     Note that some records may have more responses than expected if multiple team members submit responses on the same record simultaneously.

Reviewer 2: Also, the issue with the Draft status and later submission can be added. How is this managed? Does this have any other consequence?
